src/radio/radio-controller.ts

```diff
@@ -165,4 +166,70 @@ export class RadioController {
       }),
     ]);
   }
+
+  /**
+   * Reads data from a TCP socket.
+   * @param address  The address to open a TCP socket on.
+   * @param port  The port to open a TCP socket on.
+   * @param timeout  How long in ms to wait before timing out the request.
+   * @returns  The data read from the TCP socket.
+   */
+  public async readTcpSocket(
+    address: string,
+    port: number,
+    timeout = RADIO_TIMEOUT
+  ): Promise<smarthome.DataFlow.TcpResponse> {
+    // Open the socket.
+    const client = net.createConnection(port, address, () => {});
+    const discoveryBuffer = new Promise<smarthome.DataFlow.TcpResponse>(
+      resolve => {
+        // Resolve when data is recieved.
+        client.on('data', data => {
+          client.end();
```

Comment:
Should we wait for the 'end' event?

Reply:
Done.  For the timeout case, I replaced the `end()` call with `destroy()`, which closes the socket more immediately (without waiting for buffers, etc.)